janusgraph-codepipelines-ci/src/main/java/org/janusgraph/codepipelines/AwsCodePipelinesCi.java

```diff
@@ -0,0 +1,290 @@
+                .map(envEntry -> createCodeBuildProjectRequest(envEntry.getKey(),
+                        codeBuildServiceRoleArn,
+                        getBuildOutputArtifactName(envEntry.getValue()),
+                        ComputeType.BUILD_GENERAL1_LARGE, //TODO externalize
+                        AWS_CODEBUILD_JAVA_OPENJDK_8)
+                )
+                .forEach(codeBuild::createProject);
+
+        final int version = 1;
+        final StageDeclaration sourceStage = createSourceStageDeclaration(version);
+        final StageDeclaration buildStage = createBuildStageDeclaration(version);
+        final CreatePipelineResult pipeline = codePipeline.createPipeline(
+                new CreatePipelineRequest().withPipeline(new PipelineDeclaration()
+                        .withRoleArn(codePipelineRoleArn)
+                        .withName(pipelineName + "-pipeline")
+                        .withVersion(Integer.MAX_VALUE)
+                        .withStages(sourceStage, buildStage)
+                        .withArtifactStore(new ArtifactStore().withType(ArtifactStoreType.S3).withLocation(s3Bucket))));
+
+        Preconditions.checkNotNull(pipeline, "created pipeline was null");
+    }
+
+    private static final Option REGION_OPTION = createRequiredOneArgOption("region" /*longOpt*/, "AWS region to create build stack");
+    private static final Option BUCKET_OPTION = createRequiredOneArgOption("bucket" /*longOpt*/, "AWS S3 bucket to store artifacts");
+    private static final Option CODEPIPELINE_ROLE_ARN_OPTION =
+            createRequiredOneArgOption("codepipeline-role-arn" /*longOpt*/, "ARN of IAM role for CodePipeline to use");
+    private static final Option GITHUB_OWNER_OPTION =
+            createRequiredOneArgOption("github-owner" /*longOpt*/, "GitHub owner (organization) of the repository to build");
+    private static final Option GITHUB_REPO_OPTION = createRequiredOneArgOption("github-repo" /*longOpt*/, "GitHub repository to build");
+    private static final Option GITHUB_BRANCH_OPTION = createRequiredOneArgOption("github-branch" /*longOpt*/, "GitHub repository branch to build");
+    private static final Option GITHUB_TOKEN_OPTION =
+            createRequiredOneArgOption("github-token" /*longOpt*/, "GitHub personal access token for AWS CodeBuild to use to build");
+    private static final Option PROFILE_OPTION =
+            createRequiredOneArgOption("profile" /*longOpt*/, "AWS credential profile to use to create the stack");
+    private static final Option PIPELINES_JSON_OPTION =
+            createRequiredOneArgOption("pipelines" /*longOpt*/,"Path to JSON file containing abbreviated pipeline definitions");
+    private static final Option CODE_BUILD_SERVICE_ROLE_ARN_OPTION =
+            createRequiredOneArgOption("codebuild-role-arn" /*longOpt*/,
+                    "ARN of the service role for CodeBuild (http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html#setting-up-service-role)");
```

Comment:
Move these option definitions to the top with other field definitions (unless there's precedent elsewhere for breaking up definitions like this)